testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -814,270 +377,63 @@ abstract class HttpClientTest<REQUEST> extends InstrumentationSpecification {
-      count.times { idx ->
-        trace(idx, 4) {
-          def rootSpan = it.span(0)
-          //Traces can be in arbitrary order, let us find out the request id of the current one
-          def requestId = Integer.parseInt(rootSpan.name.substring("Parent span ".length()))
-
-          span(0) {
-            name "Parent span " + requestId
-            kind SpanKind.INTERNAL
-            hasNoParent()
-            attributes {
-              "test.request.id" requestId
-            }
-          }
-          clientSpan(it, 1, span(0), method, url)
-          serverSpan(it, 2, span(1)) {
-            it."test.request.id" requestId
-          }
-          span(3) {
-            name "child"
-            kind SpanKind.INTERNAL
-            childOf span(0)
-          }
-        }
-      }
-    }
+    expect:
+    junitTest.highConcurrencyWithCallback()
   }
 
   /**
    * Almost similar to the "high concurrency test" test above, but all requests use the same single
    * connection.
    */
   def "high concurrency test on single connection"() {
-    setup:
-    def singleConnection = createSingleConnection("localhost", server.httpPort())
+    SingleConnection singleConnection = createSingleConnection("localhost", server.httpPort())
     assumeTrue(singleConnection != null)
-    int count = 50
-    def method = 'GET'
-    def path = "/success"
-    def url = resolveAddress(path)
-
-    def latch = new CountDownLatch(1)
-    def pool = Executors.newFixedThreadPool(4)
-
-    when:
-    count.times { index ->
-      def job = {
-        latch.await()
-        runWithSpan("Parent span " + index) {
-          Span.current().setAttribute("test.request.id", index)
-          singleConnection.doRequest(path, [(SingleConnection.REQUEST_ID_HEADER): index.toString()])
-        }
-      }
-      pool.submit(job)
-    }
-    latch.countDown()
-
-    then:
-    assertTraces(count) {
-      count.times { idx ->
-        trace(idx, 3) {
-          def rootSpan = it.span(0)
-          //Traces can be in arbitrary order, let us find out the request id of the current one
-          def requestId = Integer.parseInt(rootSpan.name.substring("Parent span ".length()))
-
-          span(0) {
-            name "Parent span " + requestId
-            kind SpanKind.INTERNAL
-            hasNoParent()
-            attributes {
-              "test.request.id" requestId
-            }
-          }
-          clientSpan(it, 1, span(0), method, url)
-          serverSpan(it, 2, span(1)) {
-            it."test.request.id" requestId
-          }
-        }
-      }
-    }
+    expect:
+    junitTest.highConcurrencyOnSingleConnection()
   }
 
-  //This method should create either a single connection to the target uri or a http client
-  //which is guaranteed to use the same connection for all requests
-  SingleConnection createSingleConnection(String host, int port) {
-    return null
+  // ideally private, but then groovy closures in this class cannot find them
```

Comment:
Is this comment still valid? This class doesn't call `doRequest()`, tests extending this one call it.